sphinx/source/docs/user_guide/layout.rst

```diff
@@ -3,11 +3,13 @@
 Laying out Plots and Widgets
 ============================
 
+TODO rewrite this
```

Comment:
Checking out the branch to ensure forward compatibility of my project, I was looking for some instructions on how to make a grid plot full-width when I stumbled upon this marker.

I later managed to stretch my grid with:
```py
for fig in figures:
    fig.width_policy = 'max'

fig = gridplot(figures, ncols=1, toolbar_location='right')

# 2¢: I find this necessary part a bit too particular
fig.children[0].width_policy = 'max'
```
Looks good, though. 👍

Reply:
@kernc Just trying to make sure I fully understand: did you have something working before, and this had to change after this PR? Or you did not have anything working before?

Reply:
Worry not, what I had before is still working perfectly with this PR. :+1:  I was just trying out the adaptive 100% width my project will benefit from.

I commented, foremost, to bring attention to the remaining TODO marker in the docs. :smile:

Reply:
This is actually a bug, because `max` policy on children should expand parent container(s) if allowed (which is true in this case). It looks like this expanding flag doesn't get propagated in nested layouts (toolbar introduces a nested layout in a grid plot).

Reply:
@mattpap I suppose while at it, maybe check if height policy doesn't get propagated as well. I can't for the life of me make it work.

Reply:
@kernc, can you show a bit of code that doesn't work? It will make my life much easier if we speak about specific bits of code, which then I could turn easily into tests.

Reply:
@mattpap Sorry, season. I would expect the following plots in a grid to span full browser (container) size (width and height). Obviously, even without `p.children[0].*_policy = 'max'` lines.
```py
import numpy as np
from bokeh.io import show, output_file
from bokeh.layouts import gridplot
from bokeh.plotting import figure


output_file(filename='/tmp/bokeh-test.html', mode='absolute')

plots = []
for i in range(2):
    p = figure(
        height_policy='max',
        width_policy='max',
    )
    p.line(np.arange(10), np.random.random(10))
    plots.append(p)

p = gridplot(plots, ncols=1, toolbar_location='right',)

# As established, without this the widths are truncated to 0
p.children[0].width_policy = 'max'

# But even with this, heights weird.
p.children[0].height_policy = 'max'

show(p)
```